src/map/lua/lua_baseentity.cpp

```diff
@@ -10036,6 +10036,30 @@ void CLuaBaseEntity::recalculateAbilitiesTable()
     PChar->pushPacket(new CCharAbilitiesPacket(PChar));
 }
 
+/************************************************************************
+ *  Function: getPlayersInRange()
+ *  Purpose : Returns a Lua table of players within range of the base entity
+ *  Example : local players = npc:getPlayersInRange(50)
+ *  Notes   : if the passed argument is nil or 0, just returns all players in the zone
+ ************************************************************************/
+
+sol::table CLuaBaseEntity::getPlayersInRange(uint32 dist = 0)
+{
+    auto players = lua.create_table();
+
+    // clang-format off
+    zoneutils::GetZone(m_PBaseEntity->getZone())->ForEachChar([&](CCharEntity* PChar)
+    {
+        if (!dist || distance(PChar->loc.p, m_PBaseEntity->loc.p) < dist)
+        {
+            players.add(CLuaBaseEntity(PChar));
+        }
+    });
+    // clang-format on
```

Comment:
This will return ALL players within range, including dead ones. You do check for this in the instances I can see, but we have a whole `TargetFind` system that you should be using for this.

If/when we improve TargetFind, this block will likely be missed from that by accident. All target-finding should exist in TargetFind. It's much better to use TargetFind, even if you have to refactor it in another PR before its safe/easy/possible to use here.

Reply:
I'll probably refactor targetfind on Sunday, Eden has some tasty treats I've been meaning to take since 2020 and now is probably a good time to do that

Reply:
Interesting yea it didn't occur to me to use target find for this. Doesn't targetfind technically also loop over everyone in the zone? 

I agree it would be cleaner to use that system though. I've learned since making the PW and DI PRs that I should be breaking off these core changes to separate PRs to get the features I need for the main PR